### ISSUE TITLE
Fix unreliable Start menu replacement

### ISSUE THREAD

**Issue by srwi** [stnkl/EverythingToolbar#471]
The current implementation of the Start menu replacement is very unreliable and causing multiple problems. Ideally all keyboard input within the Start menu should be redirected to EverythingToolbar. The new implementation must fix/fulfill the following:

- [ ] #371
- [ ]  #395
- [ ]  #432
- [ ]  #448
- [ ]  #185
- [ ]  #341

Any help - either technical or just by providing ideas - is much appreciated.

**Comment by ARC345** [comment]
please fix #371

**Comment by e-t-l** [comment]
> Search key on Microsoft keyboard opens MS search and not ET #395

Is this something AutoHotKey is capable of intercepting?

**Comment by e-t-l** [comment]
Exactly. I was thinking that if AHK can rebind it, then it's something ET should be able to do natively.. Except I recognize that if that's too much work to implement, just using AHK alongside ET would be a pretty straightforward workaround for ET users

**Comment by KoBruh** [comment]
Hey SRWI, god bless you for this absolutely amazing program. I am currently just lurking and noticing many of the same issues im facing, im not entirely sure how to check of these issues have been "fixed" in a sense and this seems to be the only location open where the issues are mentioned, i cant wait for the rewrite of this specific feature!

**Comment by KoBruh** [comment]
> Exactly. I was thinking that if AHK can rebind it, then it's something ET should be able to do natively.. Except I recognize that if that's too much work to implement, just using AHK alongside ET would be a pretty straightforward workaround for ET users

I know this is a super sucky fix to this issue, but if you have everything toolbar pinned, you can open it pressing Win+(slot of pinned item) mine in this case is in slot 1 ( it ignores the windows start menu icon ) so i would click Win + 1 and it opens the toolbar no issues, and its the closest i can get to what i want to open the app with ( ideally windows + enter key )

**Comment by yfdyh000** [comment]
In 1.5.1 on Win10, I see some problems sometimes.
After replacing Start, sometimes, it captures only the first character entered, after which focus and input are moved to the rest of the windows.

**Comment by hkpD** [comment]
> In 1.5.1 on Win10, I see some problems sometimes. After replacing Start, sometimes, it captures only the first character entered, after which focus and input are moved to the rest of the windows.

Same issue here.

**Comment by benjaminaudet** [comment]
Same here, loosing focus after one character have been input

**Comment by srwi** [comment]
I tried to improve this feature with the 1.5.2 release. Could you please give some feedback on how it's working? 

If you are having problems please try to provide instructions on how to reproduce the problem and which Windows version and EverythingToolbar variant (Deskband/launcher) you are using.

**Comment by srwi** [comment]
> I know it's experimental so I am not too worried at the moment.

Yeah... It will probably never leave that stage. I honestly thought about removing the feature altogether since it's a massive headache to be honest. But given the many requests I wanted to give it another go. 😅 It really is a struggle handling the different scenarios of how Windows moves window focus around. Even the most recently focused window plays a role. And randomness... a lot of randomness...

Anyways, are you using the [deskband](https://github.com/srwi/EverythingToolbar?tab=readme-ov-file#deskband) or [search icon/launcher](https://github.com/srwi/EverythingToolbar?tab=readme-ov-file#search-icon) variant? Also, what is the last window you focused/interacted with before hitting the Windows key?

**Comment by benjaminaudet** [comment]
@srwi have you tried disabling windows key press event default handler with something like power toys keyboard remap method, then you have a total control over windows key pressed event behavior / handle.

Have you tried using hotkey detective to investigate any interference or something like that ?

Eventually, @srwi I could give you a hand on this, i'm not c# dev but i have several years of experience in C/c++ and i'm a fullstack developer. We could call and do peer-brainstorming try to figure out what exactly is the problem and how we could find a solution. 

As you wish, I'm available, text me if you're interested. Unfortunaly I cannot contribute myself since i'm not a cs developer :/ sorry.

Cya, have a good one, take care

**Comment by DavidBevi** [comment]
I too have issues, full description›#688, below a summary:

**Env**: EverythingToolbar v2.1.1.0  search icon/launcher - Everything v1.4.1 - Windows 11 Pro 25H2

- **Text** sometimes goes to start menu instead of ET's search window, and only fully closing start menu fixes it.
- **Text** sometimes get grabbed by ET's search window with missing initial characters (1 to 3).
- **Text** sometimes gets scrambled, for example "`win`fluke" rapidly typed can produce "uflke".
- **ET's window** sometimes begins opening when I type in start menu, but suddenly disappears during / just after the opening animation.
- **ET's window** sometimes appear when I press `win` to close the start menu. It randomly stays open or closes during / just after the opening animation. In some cases the start menu also reopens over ET's window.
- [_with **Explorer Patcher**'s `start menu = W10`_] **ET's window** sometimes loses focus while typing, and I cannot type in it / close it with `esc` / interact with it until I click it.

**Comment by srwi** [comment]
I think the feature is currently in a stable state, so I'll close the issue now. Please let me know when you're experiencing problems with it.

**Comment by ImGGAAVVIINN** [comment]
It does not work in start all back 3.9.23

**Comment by srwi** [comment]
> It does not work in start all back 3.9.23

It doesn't work with StartAllBack's own start menu but it works with the default Windows 11 start menu. Supporting third party start menus can unfortunately not be supported.
